client/dashboard/domains/domain-contact-details/contact-form.tsx

```diff
@@ -0,0 +1,223 @@
+			'optOutTransferLock',
+		],
+	};
+
+	const canSave = isItemValid( formData, fields, form );
+
+	return (
+		<VStack spacing={ 10 }>
+			<Notice>
+				<VStack>
+					<Text as="p">{ __( 'Provide accurate contact information' ) }</Text>
+					<Text as="p">
+						{ createInterpolateElement(
+							sprintf(
+								/* translators: %1$s: ICANN acronym */
+								__(
+									'<external>%s</external> requires accurate contact information for registrants. This information will be validated after purchase. Failure to validate your contact information will result in domain suspension.'
+								),
+								'ICANN'
+							),
+							{
+								external: (
+									<Button
+										variant="link"
+										target="_blank"
+										href="https://www.icann.org/resources/pages/contact-verification-2013-05-03-en"
+									>
+										ICANN
+									</Button>
+								),
+							}
+						) }
+					</Text>
+					<Text as="p">
+						{ __( 'Domain privacy service is included for free on applicable domains.' ) }{ ' ' }
+						<ExternalLink
+							// eslint-disable-next-line wpcalypso/i18n-unlocalized-url
+							href="https://wordpress.com/support/domains/private-domain-registration/#what-is-privacy-protection"
+						>
+							{ __( 'Learn more' ) }
+						</ExternalLink>
+						.
+					</Text>
+				</VStack>
+			</Notice>
+
+			<Card>
+				<CardBody>
+					<VStack spacing={ 4 }>
+						<DataForm< DomainContactDetails >
+							data={ formData }
+							fields={ fields }
+							form={ form }
+							onChange={ ( edits: Partial< DomainContactDetails > ) => {
+								setFormData( ( data ) => ( { ...data, ...edits } ) );
+							} }
+						/>
+						<Notice>
+							<VStack>
+								<Text as="p">
+									{ createInterpolateElement(
+										__(
+											'By clicking <strong>Save contact info</strong>, you agree to the applicable <agreementlink>Domain Registration Agreement</agreementlink> and confirm that the Transferee has agreed in writing to be bound by the same agreement. You authorize the respective registrar to act as your <agentlink>Designated Agent</agentlink>.'
+										),
+										{
+											strong: <strong />,
+											agreementlink: (
+												<a
+													// eslint-disable-next-line wpcalypso/i18n-unlocalized-url
+													href="https://wordpress.com/automattic-domain-name-registration-agreement/"
+													target="_blank"
+													rel="noopener noreferrer"
+												/>
+											),
+											agentlink: (
+												<a
+													// eslint-disable-next-line wpcalypso/i18n-unlocalized-url
+													href="https://wordpress.com/support/domains/update-contact-information/#designated-agent"
+													target="_blank"
+													rel="noopener noreferrer"
+												/>
+											),
+										}
+									) }
+								</Text>
+							</VStack>
+						</Notice>
+						<HStack justify="flex-start" spacing={ 2 }>
+							<Button
+								onClick={ () => handleSubmit( formData ) }
```

Comment:
We should use html `<form>` and put the `handleSubmit` there. See: https://github.com/Automattic/wp-calypso/blob/c5e64fbcc29b5130ddb2398fab45b64d419b62a5/client/dashboard/sites/settings-php/index.tsx#L98